firebase-installations/src/main/java/com/google/firebase/installations/remote/TokenResult.java

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.installations.remote;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+
+/** This class represents a set of values describing a FIS Auth Token Result. */
+@AutoValue
+public abstract class TokenResult {
+
+  public enum ResponseCode {
+    // Returned on success
+    OK,
+    // Auth token cannot be generated for this FID in the request. Because it is not
+    // registered/found on the FIS server. Recreate a new fid to fetch a valid auth token.
+    FID_ERROR,
+    // Refresh token in this request in not accepted by the FIS server. Either it has been blocked
+    // or changed. Recreate a new fid to fetch a valid auth token.
+    REFRESH_TOKEN_ERROR,
+  }
+
+  public boolean isSuccessful() {
+    return getResponseCode() == ResponseCode.OK;
+  }
+
+  public boolean isErrored() {
+    return getResponseCode() == ResponseCode.FID_ERROR
+        || getResponseCode() == ResponseCode.REFRESH_TOKEN_ERROR;
+  }
+
```

Comment:
Will this always be a bi-state? If yes, then I prefer only having isSuccessful()

otherwise when people read the code
if (isSuccessful)
else if (isErrored)
it's a bit confusing that is there another status?

Reply:
+1

Reply:
As discussed offline, not exactly a bi-state. Updated the method name.